modules/internal/String.chpl

```diff
@@ -1429,14 +1429,15 @@ module String {
       }
       if boundsChecking {
         if r.hasLowBound() && (!r.hasHighBound() || r.size > 0) {
-          if r.low:int < 0 then
+          if r.alignedLow:int < 0 then
             halt("range ", r, " out of bounds for string");
         }
       }
       // Loop to find whether the low and high codepoint indices
       // appear within the string.  Note the byte indices of those
       // locations, if they exist.
-      const cp_low = if r.hasLowBound() && r.low:int >= 0 then r.low:int else 0;
+      const cp_low = if r.hasLowBound() && r.alignedLow:int >= 0
+                       then r.alignedLow:int else 0;
       const cp_high = if r.hasHighBound() then r.high:int else this.len;
```

Comment:
did you miss this one?

Reply:
also there is one more in line 1399 that github doesn't let me post a comment on. So maybe I am missing a detail about the high bound?

Reply:
Uh, no... I was just moving too fast.  Thank you!  (I found one more on line 1461 as well... speaking of which, shouldn't that line read `>= cp_count` now?)

Reply:
... and `< -1` for the first check?

Recall that because of all this mess I want to overhaul `_getView` soon, so.... even if it is a bug, maybe you can leave it in there for me to find a better way to do all this. I can't really tell why that boundsChecking block is there.

Reply:
I thought you were being overly cautious, but changing it led to 37 failures so... I'm listening to you now.

Reply:
https://github.com/Cray/chapel-private/issues/919 captures all the frustration with this part of the code so far. I'll get to it soon.